nvflare/private/fed/server/job_runner.py

```diff
@@ -259,9 +257,8 @@ def run(self, fl_ctx: FLContext):
                                 self.running_jobs[run_number] = ready_job
                                 job_manager.set_status(ready_job.job_id, RunStatus.RUNNING, fl_ctx)
                             except Exception as e:
-                                run_number = fl_ctx.get_prop(FLContextKey.JOB_RUN_NUMBER)
                                 if run_number:
-                                    self._delete_run(run_number, list(client_sites.keys()), fl_ctx)
+                                    self._stop_run(run_number, fl_ctx)
```

Comment:
self._stop_run() will update the job status to RunStatus.FINISHED_ABORTED first, then it will update it to RunStatus.FAILED_TO_RUN. Better change it to update only once.

Reply:
thanks for the review,

`self._stop_run` will NOT update the job status.
It is `self.stop_run` that updates the job status to `RunStatus.FINISHED_ABORTED`

Reply:
if we want to use `self.stop_run`, when should I update the job status?

Reply:
better to use self.stop_run(), because it takes care of the  self.running_jobs scenario. just no need to do two times of the status update.

Reply:
then maybe I should add a line after this function to fire JOB_ABORTED event?

Reply:
The issue for using self._stop_run is in case the job added into self.running_jobs, then it will become a ghost never got removed.